desktop/core/src/desktop/middleware_test.py

```diff
@@ -156,3 +156,21 @@ def test_ensure_safe_redirect_middleware():
     settings.MIDDLEWARE.pop()
     for finish in done:
       finish()
+
+def test_spnego_middleware():
+  done = []
+  # settings.MIDDLEWARE.append('desktop.middleware.SpnegoMiddleware')
+  try:
+    done.append(desktop.conf.AUTH.BACKEND.set_for_testing(['desktop.auth.backend.SpnegoDjangoBackend']))
+
+    c = make_logged_in_client()
```

Comment:
Check the content of make_logged_in_client, and we can create a new one that has the params we want for the authenticate() method (i.e. no `password`).

Also FYI to even bypass https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.Client.force_login

Note: you might even want here to just use an anonymous client `c = Client()`

Reply:
Thanks, the anonymous client works perfectly